totalRP3_Extended/aura/aura.lua

```diff
@@ -0,0 +1,1009 @@
+]]--
+local MAX_AURA_UPDATE_CYCLES = 10;
+
+--[[
+	minimum time to pass until the next automated call to auraCore:Update()
+	Update calls via event do not adhere to this minimum, however Update will ensure
+	that there is at maximum one active C_Timer
+]]--
+local MIN_AURA_UPDATE_INTERVAL = 0.1;
+
+local TOOLTIP_REFRESH_INTERVAL = 0.2;
+
+--[[
+	Update frequency for dynamic auras. Since variables can change at any time, let's just update
+	auras every DYN_AURA_UPDATE_INTERVAL seconds, if they happen to display variables
+]]--
+local DYN_AURA_UPDATE_INTERVAL = 1;
+
+local INSPECT_AURA_OFFSET = 10;
+local INSPECT_AURA_HEIGHT = 36;
+local INSPECT_AURA_COL_WIDTH = 36;
+local INSPECT_AURAS_PER_COL = 8;
+
+local AURA_HEIGHT = 48;
+local AURA_WIDTH = 36;
+local AURA_MARGIN = 16;
+local AURA_MARGIN_TOP = 24;
+local AURA_ROW_LENGTH = 8;
+
+--[[ DATA MAP
+
+	            auraCore.activeAuras[i]                        TRP3_API.profile.getPlayerCurrentProfile().auras[j]
+	
+	                     |                                                              |
+	                    \|/                                                            \|/
+	
+	+-----------------------------------------------+          +------------------------------------------------+
+	| t persistent  pointer to stored part ---------|--------> | s id            full class id                  |
+	| t class       cached class data (pointer)     |          | t vars          aura variables                 |
+	| s duration    remaining duration string       |          | n expiry        expiry timestamp               |
+	| s overlay     overlay string                  |          | n dormantSince  timestamp of last deactivation |
+	| b initialize  flag to run the onInit script   |          | n lastTick      timestamp of the last onTick   |
+	| b expired     flags the aura as expired       |          | b invalid       invalidated flag (aura should  |
+	| b cancel      flag to run the onCancel script |          |                 be removed on next Update)     |
+	| b cancelled   flags the aura as cancelled     |          +------------------------------------------------+
+	| b hasDynamicDescription   whether or not the  |
+	|               description text has tags       |
+	| b hasDynamicOverlay       whether or not the  |
+	|               overlay text has tags           |
+	+-----------------------------------------------+
+
+]]--
+
+local CUSTOM_EVENTS = TRP3_API.extended.CUSTOM_EVENTS;
+
+local auraCore;
+auraCore = {
+	
+	activeAuras = {},
+	-- a more precise version of time() that includes milliseconds
+	Now = function(self) 
+		return self.baseTime + GetTime();
+	end,
+	
+	Initialize = function(self)
+		self.baseTime = time() - GetTime();
+		self.auraFramePool = CreateFramePool("Frame", TRP3_AuraFrame, "TRP3_AuraTemplate", function(pool, frame)
+			frame:Hide();
+			frame:ClearAllPoints();
+			frame.overlay:SetText("");
+			frame.duration:SetText("");
+			frame.aura = nil;
+		end);
```

Comment:
As part of the other comment about moving template script handlers to mixins - it would be sensible to also move these initializer functions to the same mixins.

Additionally there's both initialization upon acquisition from the pool here -and- also initialization on a per-aura basis when updating the aura frame.

It would make sense to merge those and add an `Init` method to the mixin for the aura template that receives an `aura` info structure as a parameter - from which it can configure its display accordingly.